collect_app/src/main/java/org/odk/collect/android/database/instances/InstanceDatabaseMigrator.java

```diff
@@ -174,12 +179,29 @@ public void createInstancesTableV6(SQLiteDatabase db) {
                 + GEOMETRY_TYPE + " text);");
     }
 
-    private void createInstancesTableV7(SQLiteDatabase db) {
+    public void createInstancesTableV7(SQLiteDatabase db) {
+        db.execSQL("CREATE TABLE IF NOT EXISTS " + INSTANCES_TABLE_NAME + " ("
+                + _ID + " integer primary key autoincrement, "
+                + DISPLAY_NAME + " text not null, "
+                + SUBMISSION_URI + " text, "
+                + CAN_EDIT_WHEN_COMPLETE + " text, "
+                + INSTANCE_FILE_PATH + " text not null, "
+                + JR_FORM_ID + " text not null, "
+                + JR_VERSION + " text, "
+                + STATUS + " text not null, "
+                + LAST_STATUS_CHANGE_DATE + " date not null, "
+                + DELETED_DATE + " date, "
+                + GEOMETRY + " text, "
+                + GEOMETRY_TYPE + " text);");
+    }
+
+    public void createInstancesTableV8(SQLiteDatabase db) {
         db.execSQL("CREATE TABLE IF NOT EXISTS " + INSTANCES_TABLE_NAME + " ("
                 + _ID + " integer primary key autoincrement, "
                 + DISPLAY_NAME + " text not null, "
                 + SUBMISSION_URI + " text, "
                 + CAN_EDIT_WHEN_COMPLETE + " text, "
+                + CAN_DELETE_BEFORE_SEND + " text, "
```

Comment:
I went with `text` here as that's what we use for `CAN_EDIT_WHEN_COMPLETE`, but I think it'd make more sense to store both of these as `integer` to save space (as SQLite doesn't have a boolean type). That feels like a change for another time though?

Reply:
I don't have strong preferences here but if you want to change anything it's better do do that now to avoid testing db upgrades twice.